frontend/src/AuthProvider.tsx

```diff
@@ -9,8 +10,37 @@ type Props = {
 }
 
 const AuthProvider = ({ authToken, setAuthToken, children }: Props) => {
-  if(!authToken || !authToken.access_token){
-    return <Login setAuthToken={setAuthToken}/>
+  function isTokenExpired(token: string) {
+    try {
+      const decodedToken: any = jwtDecode(token);
+
+      if (decodedToken && decodedToken.exp) {
+        return decodedToken.exp < Date.now() / 1000;
+      }
+    } catch (error) {
+      console.error('Error decoding JWT token:', error);
+      return true;
+    }
+  }
+
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
```

Comment:
do we need a state here?
Maybe we can directly compute the isLoggedInState from local storage. (to avoid re-render)